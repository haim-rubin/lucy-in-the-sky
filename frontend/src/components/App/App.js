import './App.css'
import Results from '../Results'
import NoResults from '../NoResults'
import { useCallback, useEffect, useState } from 'react'

const App = ({ defaultList, getImages, setFavorite }) => {

  const [ list, setList ] = useState(defaultList)


  const cbGetImages = useCallback((query) => {
     
    getImages()
      .then(( list ) => {
        setList(list)
      })
    }, [ getImages])

  const cbSetFavorite = useCallback( ({ imageId, isFavorite }) => {
      setFavorite ({ imageId, isFavorite })
      .then(({ favorites }) => {
          const nList = list.map((image) => ({
            ...image,
            isFavorite: favorites[image.imageId]
          }))
          setList([...nList])
        })
    },[list, setFavorite]
  )
  useEffect(() => {
    cbGetImages()
  }, [cbGetImages])


  return (
    <div className='app' >
      {
        (
          list.length
          ? <Results
              list={list}
              setFavorite={cbSetFavorite} />
          : <NoResults/>
        )
      }
    </div>
  );
}

export default App
