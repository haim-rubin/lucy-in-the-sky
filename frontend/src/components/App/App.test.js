import renderer from 'react-test-renderer'
import App from './App'

test('renders learn react link', () => {
  const tree =
    renderer
      .create(
        <App  defaultList={[]} getImages={() =>Promise.resolve([])}  setFavorite={() =>{}}/>
      )
      .toJSON()
  
  expect(tree).toMatchSnapshot()
})
