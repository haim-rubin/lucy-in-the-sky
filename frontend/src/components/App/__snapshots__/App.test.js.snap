// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  className="app"
>
  <div
    className="no-results"
  >
    Query doesn't produced results
  </div>
</div>
`;
